adapter/internal/discovery/xds/server.go

```diff
@@ -362,8 +362,27 @@ func UpdateAPI(vHost string, apiProject mgw.ProjectAPI, environments []string) (
 	}
 	updateVhostInternalMaps(apiYaml.ID, apiYaml.Name, apiYaml.Version, vHost, newLabels)
 
-	routes, clusters, endpoints := oasParser.GetRoutesClustersEndpoints(mgwSwagger, apiProject.UpstreamCerts,
-		apiProject.InterceptorCerts, vHost, organizationID)
+	// create cert map for API
+	certMap := make(map[string][]byte)
+	interceptCertMap := make(map[string][]byte)
+	if len(apiProject.EndpointCerts) > 0 && len(apiProject.UpstreamCerts) > 0 {
+		for url, certFile := range apiProject.EndpointCerts {
+			if certBytes, found := apiProject.UpstreamCerts[certFile]; found {
```

Comment:
what if we log anerror, if cert not found

Reply:
fixed with https://github.com/wso2/product-microgateway/pull/2402/commits/5d1eb7bb4588efe49440493e74d1d46878d3fad4